operator/controller/src/main/java/io/apicurio/registry/operator/util/BeanUtil.java

```diff
@@ -0,0 +1,38 @@
+package io.apicurio.registry.operator.util;
+
+import io.apicurio.registry.operator.OperatorException;
+import io.quarkus.arc.Arc;
+import io.quarkus.runtime.StartupEvent;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * This is needed because QOSDK does not support bean injection into dependent resource classes. We work
```

Comment:
There was a comment I found, but I can't find it anymore. I have confirmed it experimentally though. We can use singletons in this case, but when I update the code with more features, the problem will be back. We can use singletons despite this, but I expect it'll be much more inconvenient.

Reply:
Is it this one?
https://github.com/quarkiverse/quarkus-operator-sdk/issues/720

Reply:
In that case I'd suggest trying to avoid using DI even if it would be convenient, and rely on configuration and generic/static methods.

Let see how far we can go with this approach, if it's enough for other operators I assume it would be fine for us despite it being sub-optimal.

Reply:
If you insist, but I think it's worth it using CDI.

Reply:
In general, I would try to go in a direction where the discriminators are "simpler", wondering how other projects are doing ( [KC](https://github.com/search?q=repo%3Akeycloak%2Fkeycloak%20Discriminator&type=code) )